agent/grpc-external/services/resource/watch.go

```diff
@@ -1,16 +1,40 @@
 package resource
 
 import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/consul/acl"
 	"github.com/hashicorp/consul/internal/storage"
 	"github.com/hashicorp/consul/proto-public/pbresource"
 )
 
 func (s *Server) WatchList(req *pbresource.WatchListRequest, stream pbresource.ResourceService_WatchListServer) error {
 	// check type exists
-	if _, err := s.resolveType(req.Type); err != nil {
+	reg, err := s.resolveType(req.Type)
+	if err != nil {
 		return err
 	}
 
+	// check acls
+	// TODO(spatel): FIXME Where do I get the ctx in order to extract the token?
+	ctx := context.Background()
+	authz, err := s.ACLResolver.ResolveTokenAndDefaultMeta(tokenFromContext(ctx), nil, nil)
+	if err != nil {
+		return fmt.Errorf("getting authorizer: %w", err)
```

Comment:
Although gRPC will automatically treat this as an `INTERNAL` error, it might be clearer to be explicit about the status code here (and in a handful of other places I see).

```suggestion
		return status.Errorf(codes.Internal, "failed to get authorizer: %v", err)
```

Reply:
👍 Fixed in 
- https://github.com/hashicorp/consul/pull/16842/commits/ee5b8bdc94022489faf3794aaff811869999a0c0
- https://github.com/hashicorp/consul/pull/16842/commits/d998f82794d593ade6f8dc700976298d9694ee47
- https://github.com/hashicorp/consul/pull/16842/commits/add7f76e42e5b4bf951c90ab9705121be7edd2ba